Stack/Opc.Ua.Core/Security/Certificates/EccUtils.cs

```diff
@@ -126,6 +126,47 @@ public static NodeId GetEccCertificateTypeId(X509Certificate2 certificate)
             }
         }
 
+#if ECC_SUPPORT
+        /// <summary>
+        /// returns an ECCCurve if there is a matching supported curve for the provided certificate type id. if no supported ECC curve is found null is returned.
+        /// </summary>
+        /// <param name="certificateType">the  application certificatate type node id</param>
+        /// <returns>the ECCCurve, null if certificatate type id has no matching supported ECC curve</returns>
+        public static ECCurve? GetCurveFromCertificateTypeId(NodeId certificateType)
+        {
+            ECCurve? curve = null;
+
+            if (certificateType == ObjectTypeIds.EccApplicationCertificateType ||
```

Comment:
Set also the hash algorithm corresponding to the curve and return it : See https://github.com/mrsuciu/UA-.NETStandard/blob/1396fcdd028d0182b1f0153dbe71d694d851d036/Libraries/Opc.Ua.Configuration/ApplicationInstance.cs#L914